File: travelAgency/src/main/java/group/travelagency/repository/Repository.java
package group.travelagency.repository;


import group.travelagency.domain.Entity;

import java.sql.SQLException;
import java.util.List;
import java.util.Optional;

public interface Repository<ID, T extends Entity<ID>> {
    void add(T entity);
    void update (T entity);
    Optional<T> findByID(ID id) throws SQLException;
    void delete(ID id);
    List<T> getAll();
}

File: travelAgency2/travelAgency2/Repository/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.src.Domain;

namespace travelAgency2.src.Repository
{
    internal interface Repository<ID, T> where T : Entity<ID>
    {
        void add(T entity);
        void update(T entity);
        T findByID(ID id);
        void delete(ID id);
        List<T> getAll();
    }
}
